src/posix/platform/daemon.cpp

```diff
@@ -81,7 +81,8 @@ int Daemon::OutputFormatV(const char *aFormat, va_list aArguments)
 
     rval = vsnprintf(buf, sizeof(buf) - 1, aFormat, aArguments);
```

Comment:
Just curious, why it manually places an '\0' instead of let `vsnprintf` do it?

Will `rval = vsnprintf(buf, sizeof(buf), aFormat, aArguments);` do the same?

Reply:
The `vsnprintf` will add `\0` to the end of the buffer. I don't know why it manually places an '\0' to the end. @bukepo Do you know the reason?

Reply:
Yes, then the manual `\0` is not necessary.

Reply:
Updated